Game/Assets/Scripts/testCollision.cpp

```diff
@@ -0,0 +1,23 @@
+#include "testCollision.h"
+#include "GameplaySystems.h"
+
+EXPOSE_MEMBERS(testCollision) {
+    // Add members here to expose them to the engine. Example:
+    // MEMBER(MemberType::BOOL, exampleMember1),
+    // MEMBER(MemberType::PREFAB_RESOURCE_UID, exampleMember2),
+    // MEMBER(MemberType::GAME_OBJECT_UID, exampleMember3)
+};
+
+GENERATE_BODY_IMPL(testCollision);
+
+void testCollision::Start() {
+	
+}
+
+void testCollision::Update() {
+	
+}
+
+void OnCollision() {
+    Debug::Log("Collision Detected");
```

Comment:
i dont really undestand how this work couse i dont see some function in the gameplaysistem.cpp or includes

Reply:
is just to know for me .

Reply:
So, the way collision reaction is handled by scripts is analogous to that of button clicked: the Script class has an onCollision method that may be overrided by your script; collider components also have a method onCollision, which gets all the ComponentScript in their owners (if any) and call their onCollision methods in turn.

Then, the flow is: ModulePhysics, on PreUpdate, performs the stepSimulation of the world; for every contact that occurred (if any), the onCollision method of the collider components involved is called, and they call the onCollision methods of the scripts owned by their owner.

Reply:
i got it, okay, just remove the timestamp and for me its allight

Reply:
Isaac should do some QA but this is up to Jaime..

Reply:
Also, OnCollision is a virtual function in Script.h, that must be overriden in each script that need this functionality.